jax/lax/lax.py

```diff
@@ -2484,6 +2517,26 @@ def _reshape_impl(operand, new_sizes, dimensions, old_sizes):
     return xla.apply_primitive(reshape_p, operand, new_sizes=new_sizes,
                                dimensions=dimensions, old_sizes=old_sizes)
 
+def _is_singleton_reshape(old, new):
+  # A singleton reshape is one where only singleton dimensions are added. We
```

Comment:
But _removing_ singleton dimensions isn't supported yet? Can that be represented in a lexpr?

Reply:
Yes, removing isn't supported. Adding a singleton dimension can also be written as a broadcast, but removing a singleton dimension can't be iiuc.

I think perhaps we can make them work without changing lexprs, basically by ignoring/flattening them, but I left it as a "follow-up work" item in #1668.